### DISCUSSION TITLE
Press and hold alternate button function

### DISCUSSION THREAD

**Discussion by ed-apex** [shanteacontrols/OpenDeck#206] [Ideas]
Hi,

I compiled and setup OpenDeck on a Pi Pico last night.  I'm super, super impressed.  It's terrific.  There's one feature I'd really like to add and am interested in seeing how big the lift is.  It verges on simple enough that I might be able to make the changes to the source myself, though I'm so unfamiliar with QT and such a hack that I doubt I could integrate it into the configurator.  

What I'd like to do is have a press and hold button option.  On initial press, one midi command is sent, If the pedal is held for say, 2 seconds, on release it would send a different signal.  As I'm using this to replace a diy midi controller I use with amplitube, the intent is to create a button that can both send a PC message and also increment the program change offset for the OpenDeck based pedal.  Basically, have 4 foot switches, that send PC change 0-3 and if you press and hold 3 it increments the offset on the button release or if you press and hold 0 it decrements the offset.

If anyone has a starting point of what files to look at in the source code to insert my own janky code into, that would be great.  I have an implementation built on the arduino framework on an ESP32, but without the configurator tool like OpenDeck, changes are painful.  I also suspect that others would like and benefit from this feature.  

Thanks!

**Comment by paradajz** [comment]
I'm glad you like the project but what you're asking for is deeply custom functionality which I have no plans on implementing. 

I guess the relevant starting point would be this:
https://github.com/shanteacontrols/OpenDeck/blob/master/src/firmware/application/io/buttons/Buttons.cpp#L223

But note that the firmware doesn't track the button hold time so you'd need to implement that yourself.

> I'm so unfamiliar with QT and such a hack that I doubt I could integrate it into the configurator.

The configurator is written in Typescript:
https://github.com/shanteacontrols/OpenDeckUI

**Reply by ed-apex** [reply]
You know, I ended up using 12 buttons which still gives me a few to spare, so I can have dedicated buttons for bank change, bring up tuner and something else tbd.  I'm excited though as this should be way slicker than my super hacky solution that I'm using (which admittedly works and it's what I use with my band) but this will be way better.  Plus I can add LEDs super easily that have their behaviors gui configurable, so that's very cool.  Thanks for all the hardwork on this.  What a great project.  I'm actually thinking about making a rotary encoder "remote" using opendeck that's bluetooth so that I can tweak parameters without having to kneel down (lazy yes!).  If only IKMultimedia had all of the functions their proprietary pedal board offers available as midi commands!  Looks like I'm going to have to use something like autohotkey to make a macro to bring up the tuner :/
